contracts/YieldStrategyLever.sol

```diff
@@ -0,0 +1,155 @@
+    /// @param seriesId The pool (and thereby series) to borrow from.
+    /// @param vaultId The vault id to put collateral into and borrow from.
+    /// @param borrowAmount The amount of FYTOKEN borrowed in the flash loan.
+    /// @param fee The fee that will be issued by the flash loan.
+    /// @param baseAmount The amount of own collateral to supply.
+    function borrow(
+        bytes6 ilkId,
+        bytes6 seriesId,
+        bytes12 vaultId,
+        uint128 baseAmount,
+        uint256 borrowAmount,
+        uint256 fee
+    ) internal override {
+        // We have borrowed FyTokens, so sell those
+        IPool pool = IPool(LADLE.pools(seriesId));
+        IFYToken tempFyToken = pool.fyToken();
+        tempFyToken.safeTransfer(address(pool), borrowAmount - fee);
+        uint256 baseReceived = pool.sellFYToken(address(pool), 0); // Sell fyToken to get USDC/DAI/ETH
+        pool.base().transfer(
+            address(pool),
+            pool.base().balanceOf(address(this))
+        );
+        // Mint LP token & deposit to strategy
+        pool.mintWithBase(
+            address(strategies[ilkId]),
+            msg.sender,
+            uint256(
+                pool.buyFYTokenPreview(uint128((baseReceived + baseAmount) / 3))
+            ), // TODO: what should be the fyTokenToBuy
+            0,
+            type(uint256).max
+        );
+
+        // Mint strategy token
+        uint256 tokensMinted = strategies[ilkId].mint(
+            address(LADLE.joins(ilkId))
+        );
+
+        // Borrow fyToken to repay the flash loan
+        LADLE.pour(
+            vaultId,
+            address(this),
+            int128(uint128(tokensMinted)),
+            int128(uint128(borrowAmount))
+        );
+    }
+
+    /// @param ilkId The id of the ilk being invested.
+    /// @param seriesId The seriesId corresponding to the vault.
+    /// @param vaultId The vault to repay.
+    /// @param borrowAmountPlusFee The amount of fyDai/fyUsdc that we have borrowed,
+    ///     plus the fee. This should be our final balance.
+    /// @param ink The amount of collateral to retake.
+    /// @param art The debt to repay.
+    ///     slippage.
+    function repay(
+        bytes6 ilkId,
+        bytes12 vaultId,
+        bytes6 seriesId,
+        uint128 borrowAmountPlusFee,
+        uint128 ink,
+        uint128 art
+    ) internal override {
+        IPool pool = IPool(LADLE.pools(seriesId));
+        // Payback debt to get back the underlying
+        LADLE.pour(
+            vaultId,
+            address(strategies[ilkId]),
+            -int128(ink),
+            -int128(art)
+        );
+
+        // Burn strat token to get LP
+        strategies[ilkId].burn(address(pool));
+
+        // Burn LP to get base & fyToken
+        (, uint256 bases, uint256 fyTokens) = pool.burn(
+            address(pool),
+            address(this),
+            0,
+            type(uint256).max
+        );
+        // buyFyToken
```

Comment:
Assume we buy fyToken here to repay a flash loan. An ascii diagram or some comprehensive comments would be nice.